main/acle.md

```diff
@@ -1072,6 +1083,34 @@ including the header:
 
 Including `<arm_sme.h>` also includes [`<arm_sve.h>`](#arm_sve.h).
 
+### Predefined feature macros and header files
+
+Feature macros indicate whether that feature's intrinsics and inline assembly instructions
+are available in normal unannotated functions. In some implementations, the macro definitions
+are invariant throughout a translation unit, while in other implementations, the macro
+definitions can vary with pragmas.
+
+The macro definitions are otherwise independent of where the test is performed,
+meaning that a feature macro does not guarantee that the feature can be used at the
+point that it is tested. For example:
+
+``` c
+    #include <arm_sme.h>
+
+    void foo(svbool_t pg, void *ptr, uint32_t slice_base) {
+    #ifdef __ARM_FEATURE_SME
+      svst1_hor_za8(0, slice_base, pg, ptr);
+    #endif
```

Comment:
I thought the paragraph above is more about the evaluation context/order of the macro to express that a user can't rely on the macro being set/unset when used with the ACLE keyword attributes, e.g.

```
__attribute__((target("+sve")))
void foo() {
#ifdef __ARM_FEATURE_SVE
  // The user should make no assumptions that the target attribute
  // has enabled the __ARM_FEATURE_SVE macro.
#endif
}
```

The current example is more about showing that the compiler can add additional restrictions to the use of an intrinsic beyond what is captured by the ACLE macro. If that is the intent of the example, I think it would be helpful if that can be added as an separate paragraph, e.g. something like "The compiler may add additional restrictions to the intrinsics beyond what is captured by the ACLE macros depending on the context in which the intrinsics are used".

Reply:
Thank you @sdesmalen-arm  for the suggestions.
I made some changes according to your comments.